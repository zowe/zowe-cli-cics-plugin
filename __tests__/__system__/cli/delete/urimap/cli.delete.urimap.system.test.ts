/*
* This program and the accompanying materials are made available under the terms of the *
* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
* https://www.eclipse.org/legal/epl-v20.html                                      *
*                                                                                 *
* SPDX-License-Identifier: EPL-2.0                                                *
*                                                                                 *
* Copyright Contributors to the Zowe Project.                                     *
*                                                                                 *
*/

import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../../../__src__/doc/ITestPropertiesSchema";
import { generateRandomAlphaNumericString } from "../../../../__src__/TestUtils";

let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;
let regionName: string;
let csdGroup: string;
let host: string;
let port: number;
let user: string;
let password: string;
let protocol: string;
let rejectUnauthorized: boolean;
let certificate: string;
const enable: string = "false";
const authenticate: string = "BASIC";
const tcpipservice: string = "TESTSVC";

describe("CICS delete urimap command", () => {

    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "delete_urimap",
            installPlugin: true,
            tempProfileTypes: ["cics"]
        });
        csdGroup = TEST_ENVIRONMENT.systemTestProperties.cmci.csdGroup;
        regionName = TEST_ENVIRONMENT.systemTestProperties.cmci.regionName;
        host = TEST_ENVIRONMENT.systemTestProperties.cics.host;
        port = TEST_ENVIRONMENT.systemTestProperties.cics.port;
        user = TEST_ENVIRONMENT.systemTestProperties.cics.user;
        password = TEST_ENVIRONMENT.systemTestProperties.cics.password;
        protocol = TEST_ENVIRONMENT.systemTestProperties.cics.protocol;
        rejectUnauthorized = TEST_ENVIRONMENT.systemTestProperties.cics.rejectUnauthorized;
        certificate = TEST_ENVIRONMENT.systemTestProperties.urimap.certificate;
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should be able to display the help", () => {
        const output = runCliScript(__dirname + "/__scripts__/delete_urimap_help.sh", TEST_ENVIRONMENT, []);
        expect(output.stderr.toString()).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toMatchSnapshot();
    });

    it("should be able to delete a urimap of type server with basic options", async () => {
        const urimapNameSuffixLength = 3;
        const urimapName = "X" + generateRandomAlphaNumericString(urimapNameSuffixLength);
        const urimapPath = "fake/path";
        const urimapHost = "www.example.com";
        const programName = "FAKEPGM";

        let output = runCliScript(__dirname + "/../../define/urimap-server/__scripts__/define_urimap_server.sh", TEST_ENVIRONMENT,
            [urimapName, csdGroup, urimapPath, urimapHost, programName, regionName, enable, tcpipservice]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/delete_urimap.sh", TEST_ENVIRONMENT,
            [urimapName, csdGroup, regionName]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });

    it("should be able to delete a urimap of type pipeline with basic options", async () => {
        const urimapNameSuffixLength = 3;
        const urimapName = "X" + generateRandomAlphaNumericString(urimapNameSuffixLength);
        const urimapPath = "fake/path";
        const urimapHost = "www.example.com";
        const pipelineName = "FAKEPIPE";

        let output = runCliScript(__dirname + "/../../define/urimap-pipeline/__scripts__/define_urimap_pipeline.sh", TEST_ENVIRONMENT,
            [urimapName, csdGroup, urimapPath, urimapHost, pipelineName, regionName, enable, tcpipservice]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/delete_urimap.sh", TEST_ENVIRONMENT,
            [urimapName, csdGroup, regionName]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });

    it("should be able to delete a urimap of type client with basic options", async () => {
        const urimapNameSuffixLength = 3;
        const urimapName = "X" + generateRandomAlphaNumericString(urimapNameSuffixLength);
        const urimapPath = "fake/path";
        const urimapHost = "www.example.com";

        let output = runCliScript(__dirname + "/../../define/urimap-client/__scripts__/define_urimap_client.sh", TEST_ENVIRONMENT,
            [urimapName, csdGroup, urimapPath, urimapHost, regionName, enable, authenticate, certificate]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/delete_urimap.sh", TEST_ENVIRONMENT,
            [urimapName, csdGroup, regionName]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });

    it("should get a syntax error if csdGroup is omitted", () => {
        const output = runCliScript(__dirname + "/__scripts__/delete_urimap.sh", TEST_ENVIRONMENT, ["FAKENAME", ""]);
        const stderr = output.stderr.toString();
        expect(stderr).toContain("Syntax");
        expect(stderr).toContain("Missing Positional Argument");
        expect(stderr).toContain("csdGroup");
        expect(output.status).toEqual(1);
    });

    it("should get a syntax error if urimapName is omitted", () => {
        const output = runCliScript(__dirname + "/__scripts__/delete_urimap.sh", TEST_ENVIRONMENT, ["", "FAKEGRP"]);
        const stderr = output.stderr.toString();
        expect(stderr).toContain("Syntax");
        expect(stderr).toContain("Missing Positional Argument");
        expect(stderr).toContain("urimapName");
        expect(output.status).toEqual(1);
    });

    it("should be able to successfully delete a urimap of type server with profile options", async () => {

        const urimapNameSuffixLength = 3;
        const urimapName = "X" + generateRandomAlphaNumericString(urimapNameSuffixLength);
        const urimapPath = "fake/path";
        const urimapHost = "www.example.com";
        const programName = "FAKEPGM";
        const urimapScheme = "HTTPS";

        let output = runCliScript(__dirname + "/../../define/urimap-server/__scripts__/define_urimap_server_fully_qualified.sh", TEST_ENVIRONMENT,
            [urimapName,
                csdGroup,
                urimapPath,
                urimapHost,
                urimapScheme,
                programName,
                regionName,
                enable,
                tcpipservice,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/delete_urimap_fully_qualified.sh", TEST_ENVIRONMENT,
            [urimapName,
                csdGroup,
                regionName,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });

    it("should be able to successfully delete a pipeline of type server with profile options", async () => {

        const urimapNameSuffixLength = 3;
        const urimapName = "X" + generateRandomAlphaNumericString(urimapNameSuffixLength);
        const urimapPath = "fake/path";
        const urimapHost = "www.example.com";
        const pipelineName = "FAKEPIPE";
        const urimapScheme = "HTTPS";

        let output = runCliScript(__dirname + "/../../define/urimap-pipeline/__scripts__/define_urimap_pipeline_fully_qualified.sh", TEST_ENVIRONMENT,
            [urimapName,
                csdGroup,
                urimapPath,
                urimapHost,
                urimapScheme,
                pipelineName,
                regionName,
                enable,
                tcpipservice,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/delete_urimap_fully_qualified.sh", TEST_ENVIRONMENT,
            [urimapName,
                csdGroup,
                regionName,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });

    it("should be able to successfully delete a urimap of type client with profile options", async () => {

        const urimapNameSuffixLength = 3;
        const urimapName = "X" + generateRandomAlphaNumericString(urimapNameSuffixLength);
        const urimapPath = "fake/path";
        const urimapHost = "www.example.com";
        const urimapScheme = "HTTPS";

        let output = runCliScript(__dirname + "/../../define/urimap-client/__scripts__/define_urimap_client_fully_qualified.sh", TEST_ENVIRONMENT,
            [urimapName,
                csdGroup,
                urimapPath,
                urimapHost,
                urimapScheme,
                regionName,
                enable,
                authenticate,
                certificate,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        let stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");

        output = runCliScript(__dirname + "/__scripts__/delete_urimap_fully_qualified.sh", TEST_ENVIRONMENT,
            [urimapName,
                csdGroup,
                regionName,
                host,
                port,
                user,
                password,
                protocol,
                rejectUnauthorized]);
        stderr = output.stderr.toString();
        expect(stderr).toEqual("");
        expect(output.status).toEqual(0);
        expect(output.stdout.toString()).toContain("success");
    });
});
