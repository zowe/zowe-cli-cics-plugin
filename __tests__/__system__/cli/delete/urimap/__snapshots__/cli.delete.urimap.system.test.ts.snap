// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CICS delete urimap command should be able to display the help 1`] = `
"
 COMMAND NAME
 ------------

   urimap

 DESCRIPTION
 -----------

   Delete a urimap from CICS.

 USAGE
 -----

   zowe cics delete urimap <urimapName> <csdGroup> [options]

 POSITIONAL ARGUMENTS
 --------------------

   urimapName		 (string)

      The name of the urimap to delete. The maximum length of the urimap name is eight
      characters.

   csdGroup		 (string)

      The CICS system definition (CSD) Group for the urimap that you want to delete.
      The maximum length of the group name is eight characters.

 OPTIONS
 -------

   --region-name (string)

      The CICS region name from which to delete the urimap

 CICS CONNECTION OPTIONS
 -----------------------

   --host | -H (string)

      The CICS server host name.

   --port | -P (number)

      The CICS server port.

      Default value: 443

   --user | -u (string)

      Mainframe (CICS) user name, which can be the same as your TSO login.

   --password | --pw (string)

      Mainframe (CICS) password, which can be the same as your TSO password.

   --reject-unauthorized | --ru (boolean)

      Reject self-signed certificates.

      Default value: true

   --protocol | -o (string)

      Specifies CMCI protocol (http or https).

      Default value: https
      Allowed values: http, https

 PROFILE OPTIONS
 ---------------

   --cics-profile | --cics-p (string)

      The name of a (cics) profile to load for this command execution.

   --base-profile | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --token-type | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value | --tv (string)

      The value of the token to pass to the API.

   --cert-file (local file path)

      The file path to a certificate file to use for authentication

   --cert-key-file (local file path)

      The file path to a certificate key file to use for authentication

 GLOBAL OPTIONS
 --------------

   --show-inputs-only (boolean)

      Show command inputs and do not run the command

   --response-format-json | --rfj (boolean)

      Produce JSON formatted data from a command

   --help | -h (boolean)

      Display help text

   --help-web | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - Delete a urimap named URIMAPA from the region named
   MYREGION belonging to the csdgroup MYGRP:

      $ zowe cics delete urimap URIMAPA MYGRP --region-name MYREGION

"
`;
